mlflow/cli.py

```diff
@@ -304,6 +304,9 @@ def gc(backend_store_uri, run_ids):
     This command deletes all artifacts and metadata associated with the specified runs.
     """
     backend_store = _get_store(backend_store_uri, None)
+    if not hasattr(backend_store, '_hard_delete_run'):
```

Comment:
@jdlesage @smurching I'm a bit confused as to why this is necessary. The `_hard_delete_run` method is not implemented in the REST store, so this CLI would not work when targeted against the MLflow Server directly. Further, client-side changes won't address server-side gaps if they were to exist (e.g., users can simply remove these new lines of code and run `mlflow gc`). 

Regardless, I think this is a nice improvement to the UX because it raises a clear error message when an invalid store is supplied - thanks @jdlesage. Please feel free to correct me if I'm missing something here.

Reply:
@dbczumar I think you've mistakenly pinged jd.lesage ;-) 

The issue was that the error about not implemented _hard_delete_run would be thrown after having already deleted artifacts in the artifacts store. 
This check avoids to mistakenly wipe artifacts.